packages/legend-query-builder/src/components/__tests__/QueryBuilderProjectVaraints.test.tsx

```diff
@@ -0,0 +1,187 @@
+      entities: TEST_DATA__QueryBuilder_Model_SimpleRelational,
+      mapping: 'execution::RelationalMapping',
+      runtime: 'execution::Runtime',
+      rawLambda: TEST_DATA__projectWithLambdaNotCollection,
+      colLength: 1,
+      colAlias: ['Id'],
+    },
+  ],
+  [
+    'Simple projection whose col names (#2 parameter) is not a collection',
+    {
+      _class: 'model::Firm',
+      entities: TEST_DATA__QueryBuilder_Model_SimpleRelational,
+      mapping: 'execution::RelationalMapping',
+      runtime: 'execution::Runtime',
+      rawLambda: TEST_DATA__projectWithColumnNameNotCollection,
+      colLength: 1,
+      colAlias: ['Id'],
+    },
+  ],
+  [
+    'Simple projection with single col function',
+    {
+      _class: 'model::Firm',
+      entities: TEST_DATA__QueryBuilder_Model_SimpleRelational,
+      mapping: 'execution::RelationalMapping',
+      runtime: 'execution::Runtime',
+      rawLambda: TEST_DATA__projectWithSingleColFunction,
+      colLength: 1,
+      colAlias: ['Id'],
+    },
+  ],
+  [
+    'Simple projection with a col fuc collection',
+    {
+      _class: 'model::Firm',
+      entities: TEST_DATA__QueryBuilder_Model_SimpleRelational,
+      mapping: 'execution::RelationalMapping',
+      runtime: 'execution::Runtime',
+      rawLambda: TEST_DATA__projectWithColFunctionCollection,
+      colLength: 1,
+      colAlias: ['Id'],
+    },
+  ],
+];
+
+describe(
+  integrationTest('Test Query Builder is able to understand project varaints'),
+  () => {
+    test.each(cases)(
+      '%s',
+      async (testName: TestCase[0], testCase: TestCase[1]) => {
+        const {
+          _class,
+          entities,
+          mapping,
+          runtime,
+          rawLambda,
+          colLength,
+          colAlias,
+        } = testCase;
+        const { renderResult, queryBuilderState } =
+          await TEST__setUpQueryBuilder(
+            entities,
+            stub_RawLambda(),
+            mapping,
+            runtime,
+            TEST_DATA__ModelCoverageAnalysisResult_SimpleRelationalWithExists,
+          );
+
+        await act(async () => {
+          queryBuilderState.changeClass(
+            queryBuilderState.graphManagerState.graph.getClass(_class),
+          );
+        });
+        const setupPanel = await waitFor(() =>
+          renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_SETUP),
+        );
+        await waitFor(() =>
+          getByText(setupPanel, extractElementNameFromPath(_class)),
+        );
+        await waitFor(() =>
+          getByText(setupPanel, extractElementNameFromPath(mapping)),
+        );
+        expect(
+          getAllByText(setupPanel, extractElementNameFromPath(runtime)).length,
+        ).toBeGreaterThanOrEqual(1);
+        await act(async () => {
+          queryBuilderState.initializeWithQuery(
+            create_RawLambda(rawLambda.parameters, rawLambda.body),
+          );
+        });
+        const tdsState = guaranteeType(
```

Comment:
lets not test state, here, you can test that you see Columns in the panel or something like that

Reply:
it's already been tested at https://github.com/finos/legend-studio/pull/3289/files/e58c4b24982ed59f94c793a90a8c7509bb2eae35#diff-a2bbc6e04903fb0fb7bb0d3dc3b811cdd53c3777e6ab8d5d353b2be880f4c22cR178